src/node_http2.cc

```diff
@@ -821,7 +821,9 @@ void Http2Session::Send(uv_buf_t* buf, size_t length) {
   HandleScope scope(env()->isolate());
   SessionSendBuffer* req = ContainerOf(&SessionSendBuffer::buffer_, buf);
   uv_buf_t actual = uv_buf_init(buf->base, length);
-  if (stream_->DoWrite(req, &actual, 1, nullptr)) {
+  if (length == 0) {
```

Comment:
Yep, this is where I was going to be looking later on today :-) ... Essentially, after the most recent refactor, the code is a bit too eager to try sending pending data even where there's none pending. I'm going to be taking some time today resolving that but this is a great start.